processing/src/main/java/org/apache/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -254,6 +357,10 @@ protected AddToFactsResult addToFacts(
     final List<String> parseExceptionMessages = new ArrayList<>();
     final AtomicLong totalSizeInBytes = getBytesInMemory();
 
+    for (OnHeapAggregateProjection projection : aggregateProjections) {
+      projection.addToFacts(key, inputRowHolder.getRow(), parseExceptionMessages, totalSizeInBytes);
+    }
+
```

Comment:
it doesn't really make a difference either way and works fine before or after. My initial prototype was just like.. inlined through `OnHeapIncrementalIndex` and i had the projection logic first because i wanted my debugger to get to the code asap so had it at the front of the method.